### ISSUE TITLE
Code => make 80 chars width fit on a mobile screen?

### ISSUE THREAD

**Issue by slorber** [narative/gatsby-theme-novela#101]
Reading code on mobile with an horizontal scrollbar is annoying

![image](https://user-images.githubusercontent.com/749374/63843778-3fd1e900-c987-11e9-8fb5-7882fefd989c.png)

I'm using jsx code in mdx with std prettier setting and max char per line of 80. 

Unfortunately the code does not fit out of the box on mobile devices.

Shouldn't Novela ensure it renders responsively and size the font of the code editor accordingly to device size?
Otherwise, what max char per line can you officially support? I'm not against using a lower number if you give me one ;)

I'm open to any solution as long as tech articles do not end up with a scrollbar

**Comment by slorber** [comment]
no problem for me, as long as you recommend a max char per line value I'll configure my prettier to run on MDX with it

**Comment by brotzky** [comment]
Looks like the React docs break the line.
I'm not sure this is a better experience since it's quite hard to follow the code in some cases.
I think formatting your code to be narrower (60) could be a great solution

Thoughts? Wrap lines or no?

<img width="312" alt="Screen Shot 2019-08-29 at 3 47 18 PM" src="https://user-images.githubusercontent.com/8007686/63981530-5a35c080-ca74-11e9-89f7-db9052e261cc.png">

**Comment by mabasic** [comment]
It should be an option to choose. I like moving it right/left to view code, because when you have a lot of code things tend to get lost in the vertical.

**Comment by slorber** [comment]
Actually not sure to like the 60 chars limit, it's really small and makes the code less readable on desktop, even if it's better than line breaks on mobile


![image](https://user-images.githubusercontent.com/749374/64001990-52742b80-cb09-11e9-91c5-7f63c29ce020.png)

VS

![image](https://user-images.githubusercontent.com/749374/64002045-733c8100-cb09-11e9-9e11-c27f459264ae.png)

(probably not the best example but 60 chars feel quite small and only occupy a fraction of the editor width)

But on mobile:

![image](https://user-images.githubusercontent.com/749374/64002230-dc23f900-cb09-11e9-8fa4-8eb6f19e05bc.png)

I think it would be better to reduce the font size and padding because even with char 60 it would overflow. My locally modified version just fit in:

![image](https://user-images.githubusercontent.com/749374/64002526-8d2a9380-cb0a-11e9-83cb-1357b8bc8f79.png)

![image](https://user-images.githubusercontent.com/749374/64002354-1e4d3a80-cb0a-11e9-93a9-d0c7c2255152.png)

**Comment by brotzky** [comment]
It seems like wrapping the code might be the best option

**Comment by slorber** [comment]
Wrapping the code?

**Comment by brotzky** [comment]
https://github.com/narative/gatsby-theme-novela/issues/101#issuecomment-526389947

See how the code wraps here as the screen gets smaller? It breaks line (removes the `<pre /> wrap: no-wrap` flow.

**Comment by slorber** [comment]
Actually I think we should do a mix of everything ;) reco 60 chars (but it's user's choice anyway, reduce size/padding, and wrap (maybe add this as a config option?)

Maybe the best is to give control to the users.

React doc and Dan Abramov/ Kent C Dodds don't do the same thing 

![image](https://user-images.githubusercontent.com/749374/64067456-05877680-cc29-11e9-8dba-c36562b08f26.png)

![image](https://user-images.githubusercontent.com/749374/64067460-1932dd00-cc29-11e9-927c-1bef3cf7d8b7.png)

**Comment by brotzky** [comment]
We're using what Kent did in his blog, which he took form somewhere else, so it's not surprising it's the same haha :).
We remove the line numbers on mobile which I think helps.

Sounds like the solution we're leaning towards is leave it as is and possibly have some configurable styling for the code component.
